File: CSharp/Concentus/Opus/Structs/ChannelLayout.cs
using Concentus.Common.CPlusPlus;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concentus.Structs
{
    internal class ChannelLayout
    {
        internal int nb_channels;
        internal int nb_streams;
        internal int nb_coupled_streams;
        internal Pointer<byte> mapping = Pointer.Malloc<byte>(256);

        internal void Reset()
        {
            nb_channels = 0;
            nb_streams = 0;
            nb_coupled_streams = 0;
            mapping.MemSet(0, 256);
        }
    }
}

File: go/opus/ChannelLayout.go
package opus
type ChannelLayout struct {
    nb_channels        int
    nb_streams         int
    nb_coupled_streams int
    mapping            [256]int16
}

func (c *ChannelLayout) Reset() {
    c.nb_channels = 0
    c.nb_streams = 0
    c.nb_coupled_streams = 0
    c.mapping = [256]int16{}
}